pkg/spec3/benchmark_serialization_test.go

```diff
@@ -19,6 +20,29 @@ var swaggerDiffOptions = []cmp.Option{cmp.Comparer(func(a spec.Ref, b spec.Ref)
 	return a.String() == b.String()
 })}
 
+func TestOpenAPIV3RoundTrip(t *testing.T) {
+	var fuzzer *fuzz.Fuzzer
+	fuzzer = fuzz.NewWithSeed(1646791953)
+	// Make sure we have enough depth such that maps do not yield nil elements
+	fuzzer.MaxDepth(22).NilChance(0.5).NumElements(1, 7)
+	fuzzer.Funcs(OpenAPIV3FuzzFuncs...)
+	expected := &OpenAPI{}
+	fuzzer.Fuzz(expected)
+
+	j, err := json.Marshal(expected)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var actual *OpenAPI
+	err = json.Unmarshal(j, &actual)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(expected, actual) {
+		t.Fatal(cmp.Diff(expected, actual, swaggerDiffOptions...))
+	}
+}
```

Comment:
Should we also check that marshaling is stable too?

Reply:
Isn't that something that's given based on the conditions here?

We don't have a OpenAPI JSON fuzzer (and I don't think we should build one) so the base case starts from a fuzzed OpenAPI that's marshalled. The unmarshal test above proves x == Unmarshal(Marshal(x)) where x is a fuzzed OpenAPI. Writing a new test to for Unmarshal is trying to determine the relationship between Marshal(x) and Marshal(Unmarshal(Marshal(x))). Since we already showed x == Unmarshal(Marshal(x)), the latter simplifies to Marshal(x). The two would only not equal if Marshal is not stable and wouldn't exactly be a roundtrip test? 

When we implement the experimental marshaler, then it makes sense to add serialization tests to ensure that the old and new marshaler have the same behavior. WDYT?